include/picongpu/plugins/output/WriteSpeciesCommon.hpp

```diff
@@ -178,11 +175,7 @@ namespace picongpu
             typedef typename pmacc::traits::Resolve<Attribute>::type::type type;
 
             type* ptr = value.getIdentifier(Attribute()).getPointer();
-            if(ptr != nullptr)
-            {
-                __deleteArray(ptr);
-                ptr = nullptr;
-            }
+            delete[] ptr;
```

Comment:
Note: `__deleteArray(ptr)` set the pointer to `nullptr`, this part of code is now deleted. It helps a lot if a pointer is set to nullptr. In case someone is using the pointer after it is deleted you will get an exception. 
In case the pointer is not set to nullptr it is very likely that you mostly not get any error.

Reply:
In general I feel better if we set the ptr to `nullptr`.

Reply:
I know and I checked when it was used. Turned out, was never required to set to `nullptr` as either the variable went out of stack immediately, or it was a member in a class destructor